lib/_http_server.js

```diff
@@ -213,8 +213,25 @@ ServerResponse.prototype.writeHeader = function() {
 };
 
 
-function Server(requestListener) {
-  if (!(this instanceof Server)) return new Server(requestListener);
+var TLS_ARGS = ['pfx', 'key', 'passphrase', 'cert', 'ca', 'crl', 'ciphers',
```

Comment:
That would be a better solution than what is already there, but I don't think there is a global list in `tls` for options that we could hook into, so the only difference that would be made is changing the location for the list.

As sort of a counter proposal, I'd say we don't check for the TLS option names, rather just assume its TLS if there is an object, because before this it would just throw an error anyways.